console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx

```diff
@@ -12,64 +12,181 @@ import {
+    setSelectedFilterValue(value);
+    onSelectFilterValue && onSelectFilterValue(value);
+  };
+
+  const shouldDisplayChildDropdown = () => {
+    if (
+      credentialType === CredentialsType.PASSWORD ||
+      credentialType == CredentialsType.PSK ||
+      credentialType === CredentialsType.X509_CERTIFICATE
+    ) {
+      return true;
+    }
+    return false;
+  };
+
+  const getFilterOptions = () => {
+    const filterOptions = getDefaultCredentialsFiterOption(credentialType);
+    const defaultSelectedOption = shouldDisplayChildDropdown()
+      ? filterOptions[0]?.value
+      : "";
+    if (shouldDisplayChildDropdown() && credentials) {
+      const newCredentials = [...credentials];
+      newCredentials
+        ?.filter((item: any) => item?.type === credentialType)
+        .forEach((item: any) => {
+          const { "auth-id": authId } = item;
+          filterOptions.push({
+            key: authId,
+            value: authId,
+            label: `Auth ID: ${authId}`
+          });
+        });
+    }
+    setSelectedFilterValue(defaultSelectedOption);
+    setFilterOptions(shouldDisplayChildDropdown() ? filterOptions : []);
+  };
+
   return (
-    <Page id={id}>
-      <PageSection>
-        <Alert
-          variant="info"
-          isInline
-          title="Device connection configuration info"
+    <>
+      <Tooltip
+        id="ci-help-tooltip"
+        position={TooltipPosition.bottom}
+        enableFlip={false}
+        content={
+          <>
+            This info section provides a quick view of the information needed to
+            configure a device connection on the device side.
+          </>
+        }
+      >
+        <Button
+          id="ci-help-icon-button"
+          icon={<ExclamationCircleIcon />}
+          variant={ButtonVariant.link}
         >
-          This info section provides a quick view of the information needed to
-          configure a device connection on the device side.
-        </Alert>
-        <br />
-        <Grid gutter="sm">
-          <GridItem span={6}>
-            <Card>
-              <CardBody>
-                <Title size="xl" headingLevel="h1">
-                  <b>Adapters</b>
-                </Title>
-                <br />
-                <AdapterList id="ci-adapter" adapters={adapters} />
-              </CardBody>
-            </Card>
-          </GridItem>
-          <GridItem span={6}>
-            <Grid gutter="sm">
-              <GridItem>
-                <Card>
-                  <CardBody>dropdown</CardBody>
-                </Card>
-              </GridItem>
-            </Grid>
-            <CredentialsView
-              id="ci-credentials-view"
-              credentials={credentials}
-            />
-          </GridItem>
-        </Grid>
-      </PageSection>
-    </Page>
+          What is the device configuration info for?
+        </Button>
+      </Tooltip>
+      <Page id={id}>
+        <PageSection>
```

Comment:
I think, we should break this component into `<Adapters/>` and `<ConfigurationToolbar/>`

Reply:
We have already AdapterList component and this component have limited code as such don't need to over break.